app/src/main/java/com/jin/jjinweather/layer/data/location/LocationProvider.kt

```diff
@@ -1,10 +1,41 @@
 package com.jin.jjinweather.layer.data.location
 
+import android.annotation.SuppressLint
 import android.content.Context
+import android.location.Geocoder
+import android.location.LocationManager
+import com.jin.jjinweather.layer.domain.model.location.GeoPoint
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.IOException
+import java.util.Locale
 
 class LocationProvider(private val context: Context) {
-    suspend fun loadCurrentCityName(): String {
-        // 권한 획득 및 위치 정보 로직 수정될 수 있음.
-        return "서울시 방배동"
+
+    suspend fun loadCurrentCityName(latitude: Double, longitude: Double): String {
+        return withContext(Dispatchers.IO) {
+            try {
+                val geocoder = Geocoder(context, Locale.getDefault())
+                val addresses = geocoder.getFromLocation(latitude, longitude, 1)
+                if (!addresses.isNullOrEmpty()) {
+                    val address = addresses[0]
+                    "${address.adminArea ?: ""} ${address.subLocality ?: ""}".trim()
```

Comment:
제 생각에는 아마 찾고 계신것이 이것이 었을 것 같네요.
```suggestion
                    "${address.adminArea.orEmpty} ${address.subLocality.orEmpty}".trim()
```
`.orEmpty()`인지, `.orEmpty`인지 기억이 가물가물합니다.

Reply:
감사합니다........!!!!!